### PR TITLE
[Bugfix] Preserve channel-last video tensor layout

### LINKED ISSUE #5417: [Bug]: Video encoding corrupts channel-last tensors with 3 or 4 frames
### Your current environment

Current `main` at `2bb618903afc9dc29a1f29578dcac31caa047ad5`.

### Your code version

vLLM-Omni `main`.

### Describe the bug

The OpenAI-compatible video encoding path misclassifies channel-last Torch tensors when the generated video contains exactly 3 or 4 frames.

`_normalize_video_tensor()` treats every rank-4 tensor whose first dimension is 3 or 4 as `[C, F, H, W]`. However, generated video tensors are also accepted in the standard channel-last layout `[F, H, W, C]`. For a 3- or 4-frame RGB video, both the frame count and channel count are 3 or 4, so the current first-dimension-only check transposes an already channel-last tensor.

For example, a tensor with shape `[4, 2, 5, 3]` is transformed into `[2, 5, 3, 4]`. The downstream encoder then interprets it as two RGBA frames and drops the fourth channel, producing the wrong frame count and spatial shape.

This affects both synchronous MP4 encoding and fragmented MP4 streaming because both paths call `_coerce_video_to_uint8_frames()`.

Minimal reproduction:

```python
import torch
from vllm_omni.entrypoints.openai.video_api_utils import _coerce_video_to_uint8_frames

video = torch.zeros((4, 2, 5, 3), dtype=torch.uint8)
frames = _coerce_video_to_uint8_frames(video)
assert frames.shape == (4, 2, 5, 3)
```

Current result: `frames.shape == (2, 5, 3, 3)`.

Expected result: channel-last tensors preserve their frame count and shape. Channel-first `[C, F, H, W]` tensors should remain supported.

### Proposed fix

Use the shared NumPy layout normalization for Torch tensors as well. That helper already gives an explicit channel-last dimension precedence in ambiguous shapes and only converts channel-first layouts when the last dimension is not a valid channel count.

### PR DESCRIPTION
**Pull request by pxljs**
### Summary

Fix video encoding for channel-last Torch tensors containing exactly 3 or 4 frames.

Fixes #5417.

The Torch-specific normalization previously interpreted any rank-4 tensor with a leading dimension of 3 or 4 as `[C, F, H, W]`. That corrupts standard `[F, H, W, C]` tensors when `F` happens to be 3 or 4.

### Changes

- Route Torch tensors through the existing shared layout normalization logic.
- Prefer an explicit channel-last dimension for ambiguous 3/4-frame tensors.
- Add CPU regression coverage for 3-frame and 4-frame channel-last tensors.
- Add coverage preserving channel-first tensor support.

### Test plan

- `git diff --check`
- Python syntax compilation for the implementation and test modules
- Targeted pytest command: `pytest -q tests/entrypoints/openai_api/test_video_api_utils.py -k 'channel_last_video_tensor or channel_first_video_tensor'`

### Test results

All tests passed!

### CODE REVIEW

**Review by Gaohan123 — APPROVED** [review]
LGTM. Thanks
